### PR TITLE
Add on_consistent_failure callback

### PR DESCRIPTION
**Pull request by staugaard**
Added an `on_consistent_failure` callback that gets called when a test keeps failing

### CODE REVIEW

File: lib/minitest/retry.rb [hunk 1]
```diff
@@ -66,7 +75,12 @@ def run_one_method(klass, method_name)
             end
 
             result = super(klass, method_name)
-            break if result.failures.empty?
+
+            if result.failures.empty?
+              break
+            elsif Minitest::Retry.consistent_failure_callback && count == (Minitest::Retry.retry_count - 1)
```
Comment (y-yagi) [inline lib/minitest/retry.rb]: It looks like do not check in the retry loop. How about it as below? 

```diff 
-            if result.failures.empty?
-              break
-            elsif Minitest::Retry.consistent_failure_callback && count == (Minitest::Retry.retry_count - 1)
-              Minitest::Retry.consistent_failure_callback.call(klass, method_name)
-            end
+            break if result.failures.empty?
           end
         end
+
+        if Minitest::Retry.consistent_failure_callback && !result.failures.empty?
+          Minitest::Retry.consistent_failure_callback.call(klass, method_name)
+        end
```

**Comment by y-yagi** [comment]
Thanks for your PR! I like this approach. Please check my comment :smiley_cat:

File: lib/minitest/retry.rb (hunk 1 above)
Comment (staugaard) [inline lib/minitest/retry.rb]: Yeah that sounds good. I didn't realize that I hav access to `result` outside of the loop. I see that now.

Update coming.

**Comment by staugaard** [comment]
I updated the PR with your suggestions. Please have another look.

**Comment by staugaard** [comment]
I ran the test suite three times in a row on the last build. 🍏, 📗, and 💚

**Comment by staugaard** [comment]
Sorry, the comment I just deleted was intended for a different PR.

**Comment by y-yagi** [comment]
Thanks!

**Comment by y-yagi** [comment]
Released as v0.1.9

**Pull request merged**